joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50   

# whole_body_controller:
#   type: position_controllers/JointGroupPositionController
#   joints:
#     - r_shoulder_pitch
#     - r_shoulder_roll
#     - r_elbow_roll
#     - l_shoulder_pitch
#     - l_shoulder_roll
#     - l_elbow_roll
#     - r_footjoint_roll
#     - r_footjoint_pitch
#     - r_b_footjoint_pitch
#     - r_uf_knee
#     - r_df_knee
#     - r_db_knee
#     - r_ankle_pitch
#     - r_ankle_roll
#     - l_footjoint_roll
#     - l_footjoint_pitch
#     - l_uf_knee
#     - r_b_footjoint_pitch
#     - l_df_knee
#     - l_db_knee
#     - l_ankle_pitch
#     - l_ankle_roll

whole_body_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - r_shoulder_pitch
    - r_shoulder_roll
    - r_elbow_roll
    - l_shoulder_pitch
    - l_shoulder_roll
    - l_elbow_roll
    - r_footjoint_roll
    - r_footjoint_pitch
    # - r_b_footjoint_pitch
    # - r_uf_knee
    # - r_df_knee
    # - r_db_knee
    - r_ankle_pitch
    - r_ankle_roll
    - l_footjoint_roll
    - l_footjoint_pitch
    # - r_b_footjoint_pitch
    # - l_uf_knee
    # - l_df_knee
    # - l_db_knee
    - l_ankle_pitch
    - l_ankle_roll
  gains:
    r_shoulder_pitch: {p: 10.0, i: 0.1, d: 0.1, i_clamp: 0.1}
    r_shoulder_roll: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    r_elbow_roll: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    l_shoulder_pitch: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    l_shoulder_roll: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    l_elbow_roll: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    r_footjoint_roll: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    r_footjoint_pitch: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    # r_b_footjoint_pitch: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    # r_uf_knee: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    # r_df_knee: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    # r_db_knee: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    r_ankle_pitch: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    r_ankle_roll: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    l_footjoint_roll: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    l_footjoint_pitch: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    # r_b_footjoint_pitch: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    # l_uf_knee: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    # l_df_knee: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    # l_db_knee: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    l_ankle_pitch: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
    l_ankle_roll: {p: 10.0, i: 0.01, d: 0.01, i_clamp: 0.1}
